src/app-logic/web-channel.js

```diff
@@ -60,20 +121,55 @@ export async function queryIsMenuButtonEnabled(): Promise<boolean> {
  * Enable the profiler menu button.
  */
 export async function enableMenuButton(): Promise<void> {
-  type ExpectedResponse = $PropertyType<
-    MessageFromBrowserObject,
-    'ENABLE_MENU_BUTTON_DONE'
-  >;
-
-  await _sendMessageWithResponse<ExpectedResponse>({
+  await _sendMessageWithResponse({
     type: 'ENABLE_MENU_BUTTON',
-    requestId: _requestId++,
   });
 
   // The response does not return any additional information other than we know
   // the request was handled.
 }
 
+/**
+ * Ask the browser if the web channel supports getting the profile and symbolication.
+ */
+export async function querySupportsGetProfileAndSymbolicationViaWebChannel(): Promise<boolean> {
+  const response = await _sendMessageWithResponse({
+    type: 'STATUS_QUERY',
+  });
+
+  return response.version ? response.version >= 1 : false;
+}
+
+export async function getSymbolTableViaWebChannel(
+  debugName: string,
+  breakpadId: string
+): Promise<SymbolTableAsTuple> {
+  return _sendMessageWithResponse({
+    type: 'GET_SYMBOL_TABLE',
+    debugName,
+    breakpadId,
+  });
+}
+
+export async function getProfileViaWebChannel(): Promise<
+  ArrayBuffer | MixedObject
+> {
+  return _sendMessageWithResponse({
+    type: 'GET_PROFILE',
+  });
+}
+
+export async function querySymbolicationApiViaWebChannel(
```

Comment:
I realize this function isn't called -- is that planned for later ?

Reply:
That's right, it's just a small piece of preparation for #1516.